### PR TITLE
Merge the NSURLSession functionality back into master

### PR DESCRIPTION
**Pull request by boxtcarpel**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @boxtcarpel has signed the CLA. Thanks for the pull request!

**Comment by thuyBox** [comment]
+2!

File: BoxContentSDKSampleApp/Podfile.lock [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 
 SPEC CHECKSUMS:
-  box-ios-sdk: a34a7c130e83e3d2380d2f6c1f16e589ba7788ba
+  box-ios-sdk: 8f8663807e26fd5e6bb89a07c0402d2bef605d30
 
-PODFILE CHECKSUM: 321973ea596897514e372fa19af769c896c0fe55
+PODFILE CHECKSUM: 665c09eb2d2e131c961508f8a52594e06e23dab9
 
-COCOAPODS: 1.0.0
+COCOAPODS: 1.2.0
```
Comment (irvingruan) [inline BoxContentSDKSampleApp/Podfile.lock:21]: I think we want CocoaPods 1.0.0 here.

`pod _1.0.0_ install` should do the trick.

**Comment by irvingruan** [comment]
+2 after switching to 1.0.0 from 1.2.0

File: BoxContentSDKSampleApp/Podfile.lock (hunk 1 above)
Comment (boxtcarpel) [reply BoxContentSDKSampleApp/Podfile.lock:21]: 1.2.0 is the current latest stable version of CocoaPods. It'll be best for devs who are keeping up with the latest.

**Pull request merged**